flask/app/datasets.py

```diff
@@ -137,24 +140,28 @@ def list_datasets(page: int, limit: int) -> Response:
         user_id = request.args.get("user")
     else:
         user_id = current_user.user_id
+
+    query = (
+        models.Dataset.query.options(
+            contains_eager(models.Dataset.tissue_sample)
+            .contains_eager(models.TissueSample.participant)
+            .contains_eager(models.Participant.family),
+            contains_eager(models.Dataset.tissue_sample)
+            .contains_eager(models.TissueSample.participant)
+            .joinedload(models.Participant.institution),
+            contains_eager(models.Dataset.files),
+            contains_eager(models.Dataset.updated_by),
+        )
+        .join(models.Dataset.tissue_sample)
+        .join(models.TissueSample.participant)
+        .join(models.Participant.family)
+        .outerjoin(models.Dataset.files)
+        .join(models.Dataset.updated_by)
+    )
+
```

Comment:
```suggestion
    query = (
        models.Dataset.query.options(
            #tell the ORM that our join contains models we'd like to eager load, so it doesn't try to load them lazily
            #in my test, this sped up loading time by ~3x
            contains_eager(models.Dataset.tissue_sample)
            .contains_eager(models.TissueSample.participant)
            .contains_eager(models.Participant.family),
            #eager load groups and files for speed, since we're not joining here, we can't order or search by these fields
            #ordering is probably not a great loss b/c one-to-many makes it tough,
            # but if we want to search, a solution might be a subquery (something similar to the filter in analyses.py:194-208)
            # FE will have to be adjusted to disallow sorting/filtering on these fields
            selectinload(models.Dataset.groups),
            selectinload(models.Dataset.files),
        )
        # join with these since they are 1-to-1 and we want to order/filter on some of their fields
        .join(models.Dataset.tissue_sample)
        .join(models.TissueSample.participant)
        .join(models.Participant.family)
        .join(models.Dataset.updated_by)
    )

```